taglib/fileref.cpp

```diff
@@ -54,6 +54,62 @@
 
 using namespace TagLib;
 
+template <typename T>
+static File* createInternal(T arg, const String& ext, bool readAudioProperties,
```

Comment:
I thought of that, but it makes me pass filename twice for one version, and iostream, filename for the other... That feels weird, so I refrained myself.
There's a way to guess the filename based on the type that is provided, but I'm not sure that would improve readability...
If you're fine with passing filename twice I'll add it